### PR TITLE
Rails 4 2

### PR DESCRIPTION
**Pull request by jedschneider**
updating gem to support Rails 4.2 and modern versions of development libraries like Cucumber and RSpec

### CODE REVIEW

**Comment by jedschneider** [comment]
@jejacks0n if you have a few I wanted to just run a couple things by you regarding mercury and teaspoon. 
- In ea5c2f2af0cc69c3ead57f3bc90ada51f223985c I removed some messages that would be printed if the selector failed because `:message` is no longer a supported option. What do you think about just letting the default Capybara ElementNotFound error raise? Not enough info?
- In cd5af477c8a3f022af40012e86d8ef7c0e53057d I updated some of the mercury selectors to be more specific. I needed to do this to prevent capybara from raising an error when it found more than one similar element. Just wanted you to take a glance and make sure those selector seemed reasonable.
- In 48d73746de12fbeba9c0e1bf7b5b6775d8602f71 I tried to move all the teaspoon configuration over, but there are no javascript specs being found. It seems like the matchers are right, but could you double check the configuration and see if you can see anything. Getting `0 examples, 0 failures` with `rake teaspoon`. 

Thanks!

**Comment by jedschneider** [comment]
there should be 64 passing JS specs, for reference.

**Comment by jedschneider** [comment]
@gvarela the newer rspec mock and stub api is a bit more restrictive so I had to remove some expectations that under the old syntax were just not being called. please take a look at  da67513 and see if there are any places I removed something that for some reason is of value.

**Comment by jedschneider** [comment]
@gvarela also like your opinion on the `Gemfile.lock` file. it is currently not tracked in git. Not sure if this is a desired thing in a gem or not.

File: spec/models/path_spec.rb [hunk 1]
```diff
@@ -94,43 +93,57 @@
...
+        end
+
+        it 'removes front slashes' do
+          expect(request).to receive(:params).and_return(path: '//path').at_least(:once)
+          expect(Cmsimple::Path.from_request(request).destination.uri).to eq('/some-other-path')
+        end
+
+        it 'removes trailing slashes' do
+        expect(request).to receive(:params).and_return(path: '//path/').at_least(:once)
+        expect(Cmsimple::Path.from_request(request).destination.uri).to eq('/some-other-path')
```
Comment (gvarela) [inline spec/models/path_spec.rb:131]: Indentation

**Comment by gvarela** [comment]
This looks good.

Yeah you never check in a Gemfile.lock for a gem. Only for an application.

**Comment by jedschneider** [comment]
@gvarela could I get your eyes on these last couple commits, it seems like in rails 4.2 `render nothing: true` acted differently for us, where it just didn't render the layout at all so, now just calling render with an empty string. have a better suggestion? Then you guys have some overrides for responders to insert messages and handle updates appropriately for mercury. the new responders gem doesn't take a parameter for an internal method called `api_behavior` and it doesn't have the `resourceful?` instance method, I don't think. Thoughts on this approach? I'm thinking its not ideal. But out of ideas at the moment.

**Comment by jedschneider** [comment]
@jejacks0n we're on 0.9.1 for teaspoon.

**Comment by jejacks0n** [comment]
@jedschneider cool. Can you just call `head :ok` if you want to render nothing?

**Pull request closed without merging**